### ISSUE TITLE
Unable to use custom font style [v2.0.1]

### ISSUE THREAD

**Issue by dmitriifrlv** [dompdf/dompdf#3082]
### Description

The custom font style is not applied when is loaded locally or from a remote resource.  When I try to use locally hosted fonts, it doesn't apply to the pdf file. 

When I try to use it from the remote resource,  I get the error:
<img width="1440" alt="Screenshot 2022-12-13 at 17 48 13" src="https://user-images.githubusercontent.com/65853205/207461725-74551572-4824-43ec-861e-9345012772f7.png">

I also ran the example from the docs locally - https://github.com/dompdf/dompdf/wiki/About-Fonts-and-Character-Encoding and got the same `Path cannot be empty ` error 

### Code:
[Source code repo](https://github.com/dmitriifrlv/customFontError)
PHP:
```
$options = new Options;
$options->setChroot(__DIR__);
$options->isRemoteEnabled(true);
$options->setFontDir(__DIR__);
$options->setFontCache(__DIR__);
$options->setTempDir(__DIR__);
$options->setIsRemoteEnabled(true);

$dompdf = new Dompdf($options);
```

Template:
```
<html>

<head>

  <style type="text/css">
    @font-face {
      font-family: "Montserrat";
      font-weight: 400;
      src: local("Montserrat"), url(/assests/fonts/Montserrat-Regular.ttf) format("truetype");
    }

    /* @font-face {
      font-family: "Montserrat";
      font-weight: 400;
      src: local("Montserrat"), url(https://fonts.googleapis.com/css2?family=Montserrat) format("truetype");
    } */

    body {
      font-family: 'Montserrat';
    }


    p {
      font-family: 'Montserrat';
    }
  </style>

</head>

<body>
  <p>Hello World</p>
</body>

</html>
```

I also tried to run `php load_font.php ...` command as it was suggested here https://stackoverflow.com/questions/12581156/custom-fonts-for-dompdf but got the error- `Could not open input file: load_font.php`

**Comment by bsweeney** [comment]
The URL you're using is actually pointing to Google's font stylesheet and not the font file. You can see that in the information provided by the stack trace, and also going directly to https://fonts.googleapis.com/css2?family=Montserrat. Try the following instead:

```
<html>

<head>

  <link rel="preconnect" href="https://fonts.googleapis.com">
  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
  <link href="https://fonts.googleapis.com/css2?family=Montserrat" rel="stylesheet"> 

  <style type="text/css">
    body {
      font-family: 'Montserrat';
    }


    p {
      font-family: 'Montserrat';
    }
  </style>

</head>

<body>
  <p>Hello World</p>
</body>

</html>
```

I'm not sure why Dompdf is throwing an error on your system, even with the bad URL I would expect Dompdf to just reject the font. That's what happens when I test.

---

FYI, load_font.php is not included with the core Dompdf project anymore. You can get it from the [utils projects](https://github.com/dompdf/utils), though I haven't ensured it runs against recent versions of Dompdf.

**Comment by dmitriifrlv** [comment]
@bsweeney Thank you for your response!

> The URL you're using is actually pointing to Google's font stylesheet and not the font file.
It does point to the font file. The font-face which points to Google's font stylesheet is commented out in the snippet of code. Also in my comment above the screenshot, I wrote that I got this error when I use the font from a **remote**  source. When I point the URL to locally hosted fonts, I don't receive errors but the font isn't applied. 

> Try the following instead:

Thanks, I've tried to use this approach but it doesn't have any effect - the font wasn't applied. 

> FYI, load_font.php is not included with the core Dompdf project anymore. 

Before I ran the command `php load_font.php ... ` I had installed the load font package but it didn't help either. 

@bsweeney I'd greatly appreciate it if you reopen the issue because the problem is still there.

**Comment by bsweeney** [comment]
I do realize that the Google font reference is commented out. However, the stack trace you posted appears to indicate you were attempting to pull the font from Google. And were not doing so correctly (per the line that reads `file_put_contents('', '@font-face{\n...')`) and so that's the error I was addressing. If you get a different error using that technique it would be helpful to know what it is.

Also regarding that particular error, there's no file path so you should make sure the process running Dompdf has read/write access to `__DIR__` (the current working directory). You might want to be more explicit and specify the actual path.

Two points related to your local file reference:
- Using a leading "/" for a local file path is saying to Dompdf to look starting from the root of the filesystem. If the assets directory is not in your filesystem root then your local font file won't be found.
- The chroot option needs to specify a parent directories of any resource from the local filesystem that your document references.

**Comment by dmitriifrlv** [comment]
I fixed the issue#1 from the post above. The problem was that I called `isRemoteEnabled(true)` rather than `setIsRemoteEnabled(true)`.  The other issues have something to do with the routing rather than with dompdf.
